libvast/src/system/configuration.cpp

```diff
@@ -261,8 +261,13 @@ caf::error configuration::parse(int argc, char** argv) {
         return x.error();
     } else {
       // If the option is not relevant to CAF's custom options, we just store
-      // the value directly in the content.
-      put(content, key, value);
+      // the value directly in the content, or append if it's a list.
+      if (auto content_list
+          = caf::get_if<caf::config_value::list>(&content, key)) {
+        content_list->emplace_back(value);
```

Comment:
Do we need a special case here if `value` is also a list, so avoid creating a nested list?

Also, I'm not 100% sure we want to unconditionally merge here, wouldn't this imply that if there is e.g. `plugins: all` configured in some config file, there is no way to override this to *not* load all plugins from the command line?

Reply:
We handle all four possible combinations now, see the last commit.

Reply:
What about the second question? I see there is e.g. a separate `policy::merge_lists` parameter in `detail::merge_settings`, whereas here we seem to merge unconditionally. Is that intentional? (honest question, I'm not 100% sure off the top of my head how the two are related)

Reply:
> Also, I'm not 100% sure we want to unconditionally merge here, wouldn't this imply that if there is e.g. `plugins: all` configured in some config file, there is no way to override this to _not_ load all plugins from the command line?

There already is no way to override this because we merge lists from configuration files.